libraries/AP_HAL_ChibiOS/SPIDevice.cpp

```diff
@@ -380,11 +381,24 @@ bool SPIDevice::acquire_bus(bool set, bool skip_cs)
         bus.spicfg.cr1 = (uint16_t)(freq_flag | device_desc.mode);
         bus.spicfg.cr2 = 0;
 #endif
+        // read current pin mode of sck
+        const iomode_t sck_mode = palReadLineMode(spi_devices[device_desc.bus].sck_line);
+        if (device_desc.mode == SPIDEV_MODE0 || device_desc.mode == SPIDEV_MODE1) {
+            // Clock polarity is 0, so we need to set the clock line low before spi reset
+            palClearLine(spi_devices[device_desc.bus].sck_line);
+            palSetLineMode(spi_devices[device_desc.bus].sck_line, PAL_MODE_OUTPUT_PUSHPULL);
```

Comment:
minor tweak, the palSetLineMode can be outside the if()

Reply:
This shouldn't be needed in spiStop under dma_deallocate, as that is after the transaction has finished and very unlikely to cause issue.

Reply:
it is needed, as it impacts the next transaction

Reply:
ok adding